darts/timeseries.py

```diff
@@ -1469,28 +1815,23 @@ def __rmul__(self, other):
 
     def __pow__(self, n):
         if isinstance(n, (int, float, np.integer)):
-            if n < 0 and not all(self.values() != 0):
-                raise_log(ZeroDivisionError('Cannot divide by a TimeSeries with a value 0.'), logger)
-
-            new_series = self._df ** float(n)
-            return TimeSeries(new_series, self.freq_str())
+            raise_if(n < 0, 'Attempted to raise a series to a negative power.', logger)
+            return TimeSeries(self._xa ** float(n))
+        if isinstance(n, (TimeSeries, xr.DataArray, np.ndarray)):
```

Comment:
What's going on here? One time series to the power of another? Is there a use case for that? :D 
Also is it checked that they match in shape?

Reply:
You're right it might be a bit over the top and not really used... It doesn't really hurt though so I would leave it (and rely on the underlying Numpy ** function to check the sizes).